sled-agent/src/rack_setup/service.rs

```diff
@@ -1188,7 +1188,10 @@ impl ServiceInner {
             v6_routes: HashMap::new(),
         };
         let link_id = LinkId(0);
-        let addr = IpAddr::V4(uplink_config.uplink_ip);
+        // TODO We're discarding the `uplink_cidr.prefix()` here and only using
+        // the IP address; this needs to be fixed as part of
+        // https://github.com/oxidecomputer/omicron/issues/3588.
+        let addr = IpAddr::V4(uplink_config.uplink_cidr.ip());
```

Comment:
This is the second place that warrants an extra look: we're discarding the prefix here. I think this roughly where we'll need to _not_ do that in order to fully solve #3588?

Reply:
tl;dr; just the IP is fine here.

Based on my understanding of our conversation today, these addresses are going to two places now.

1. To `dpd` (what's being done here)
2. To the SMF properties of a new service being created right now.

For (1) `dpd` does not care about prefix lengths. It just uses addresses to inspect incoming packets to see if they need to go to the scrimlet e.g., determining if the destination address on a packet is owned by the switch. This has nothing to do with the prefix length. Place (2) is where we'll need the prefix length as that is how the address will get on illumos - and since illumos will be sourcing traffic from that address and resolving other addresses on the same L2 domain, the prefix length becomes important.